### ISSUE TITLE
Continuous refueling of vehicles at gas tanks using a hose.

### ISSUE THREAD

**Issue by paradigmArchitect** [DayZMod/DayZ#393]
I thought of this while running back and forth with jerry cans between a helicopter and a gas tank. After half an hour of this, I noticed the hose in my inventory, and imagined the polished and refined experience it could bring to refueling.

Many custom servers a while back (I'm happy to stick with vanilla now,) advertised "auto-refuel" in the server name, and it was easy to see by the difference in player populations between those that didn't, that this is a desired feature.

I won't be much help on the technical part, but there is lots of documentation on implementing auto-refuel. I suggest needing to have the vehicle 10 meters away from the gas tank, possibly enforcing with something similar to how when your building, if you stray to far from where you started it will fail.

Hopefully this can be a quick addition without problems, I think people would love it!

**Comment by KPanicEKS** [comment]
I really don't think auto refueling belongs in vanilla. It goes against the
core philosophy that this is a true apocalypse (no electricity) if anything
I would say add in a manual pump that was as rare as nvgs and padlocks. The
pump could be used with duct tape and 2 hoses to have a craftable working
pump and you'd have to get out and use it with a vehicle. And lastly make
it slow,that way you aren't fully refueled and ready to go like an Indy pit
crew.

**Comment by ghost** [comment]
Please, no auto-refuelling. I say, to keep the mod alive, make it even harder to maintain a car/truck. Make it a challenge and a group effort to fix a chopper. A "manual pump" is a great idea. 
- a generator (is the model already included? I think so)
- a hose
- park next to the pump, gain more range to the vehicle with multiple hoses (maybe 5m per hose, so you need 2-3 hoses to reach a chopper next to a pump/tank...)
- fill genererator with fuel from jerry can
- fire up generator (A LOT OF NOIZE, attracts zombies and players)
- fill up chopper in 2 minutes

**Comment by R4Z0R49** [comment]
I dont think we will be adding any type of auto fuel system however we did think about a manual pump system where your char has to build a pump between the vehicle and helicopter then pump fuel into the chopper. If the player walks away from the pump the fuel stops sort of thing.

**Comment by paradigmArchitect** [comment]
The context for this idea has changed, due to gas tanks having a limited supply of fuel now. I would like it to be known that I consider myself a hardcore survival player who values authentic social engagements amidst each players struggle for survival and desire to overcome the harsh environment.

In accordance with my interpretation of the spirit of DayZ, I would never suggest auto refueling as it is known from mods which value ease of use and shallow accessibility, rather than constructing a challenging and hostile environment that doesn't sacrifice on immersion.

The idea originally came from a simple pondering of how siphoning fuel OUT of a vehicle into a container was done, and why it was limited to _vehicle -> fuel can_.

Why should it not also support moving fuel between: 

1.  fuel tank -> vehicle (original idea, how it would likely be done in real life if one drove up to a tank)
2.  fuel tank -> fuel can (possibly required when fuel is below a threshold?)
3.  fuel tank -> nearby fuel tank (if a system was put in place to support defining source and destination)
4.  vehicle -> vehicle (encouraging "drive by siphoning" or feeding a vampiric helicopter from car to car)
5.  vehicle -> fuel tank (if you wanted to prep a fuel tank to quickly fill a valued vehicle)

I don't suggest requiring a hose for moving fuel from can to tank, or can to vehicle (unless needing a hose is realistic for fueling some special vehicle represented in DayZ.) I also don't want to make it seem likes any of these changes are important. I wrote this post originally with no experience with server that supported auto-refuel, and I therefore wrongly imagined it as something that would ADD depth and complexity to environmental interaction, not aid in the kind of automation that serves to separate the player from the environment.

TL;DR: I was a noob when I made this post, and I never want vanilla to lose its depth to easiness or its complexity to simplicity. I do not consider expanding the siphoning system or creating additional uses for the hose a priority for DayZ--but if it can be applied in a way that adds realistic complexity and depth to the interactive environment, without diminishing the immersive struggle of post-apocalyptic vehicle ownership, then I would support additional siphoning features.


Thank you for letting me waste your time,

PA
International Hotel, Chernogorsk
